pages/cartPage.js

```diff
@@ -0,0 +1,296 @@
+
+  //to sync the prop to state when re-rendering happens,
+  //you need a useEffect inside the child that listens to prop change:
+
+  // useEffect(() => {
+  //   cookie.set('cart', itemsInCart);
+  // }, [itemsInCart]); ////this gets run every time the value is changed, so it stores new value
+
+  // const [pieces, setPieces] = useState(0);
+  // const [itemTotal, setItemTotal] = useState(0); //yes haha
+  //here you cant go like itemmm
+  const totalCart = totalSum(itemsInCart);
+  cookie.set('total', totalCart);
+  //       const totalCart =
+  //   itemsInCart.length !== 0
+  //     ? itemsInCart.reduce((acc, cur) => {
+  //         return acc + cur.price;
+  //       }, 0)
+  //     : 0;
+  // cookie.set('total', totalCart);
+  // setTotal(totalCart);
+  // cookie.set('total', total);
+
+  // useEffect(() => {
+  //   cookie.set('total', total);
+  // }, [total]);
+
+  return (
+    <div>
+      <Head>
+        <title>Bist du Teppich: Your Cart </title>
+        <link rel="icon" href="/favicon.png" />
+      </Head>
+      <Header />
+      <div className="container">
+        <h1>Shopping Cart</h1>
+        <Link href="/products">
+          <a className="backToShop"> Back to shop</a>
+        </Link>
+        <div className="tableItems">
+          <div className="headings">
+            <h4> Product</h4>
+            <h4>Description</h4>
+            <h4>Quantity</h4>
+            <h4>Price</h4>
+            <h4>&nbsp;</h4>
+          </div>
+          {itemsInCart.length !== 0
+            ? itemsInCart.map((item) => {
+                return (
+                  <div data-cy={'item-cart'} className="item" key={item.id}>
+                    <img src={item.img} alt="item" />
+                    <p>{item.name}</p>
+                    <span data-cy={'amount-cart'} className="buttonz">
+                      <ReduceOneItem
+                        item={item}
+                        cart={cart}
+                        products={products}
+                      />
+
+                      <p className="amount-cart">{item.amount}</p>
+
+                      <AddOneItem
+                        item={item}
+                        cart={cart}
+                        products={products}
+                        itemsInCart={itemsInCart}
+                      />
+                    </span>
+                    <p className="total-cart">{item.price}€</p>
+
+                    <RemoveFromCart item={item} itemsInCart={itemsInCart} />
+                  </div>
+                );
+              })
+            : 'The cart is empty....'}
+        </div>
+        <p className="total">
+          Total:
+          {totalSum(itemsInCart)}€<br></br>
+          {itemsInCart.length !== 0 ? (
+            <Link href="/payment">
+              <a data-cy="checkout-button">
+                <button>Proceed to checkout</button>
+              </a>
+            </Link>
+          ) : (
+            ' '
```

Comment:
Same thing as before with the "considering putting the short ternary case before the long one" feedback.